user/dao/user-dao.go

```diff
@@ -35,34 +37,40 @@ type UserUpdateRequest struct {
 	Name string `valid:"type(string),required,stringlength(2|255)"`
 }
 
-func executeQueryWithRowResponse(query string, args ...interface{}) (*sql.Row, error) {
-	db, err := sql.Open("postgres", connStr)
-	if err != nil {
-		return nil, err
-	}
-	defer db.Close()
+// Executes the query, returning the row
+func executeQueryWithRowResponse(db *sql.DB, query string, args ...interface{}) (*sql.Row, error) {
 	return db.QueryRow(query, args...), nil
 }
 
 // Executes a query, returning the number of rows affected
-func executeQuery(query string, args ...interface{}) (int64, error) {
-	db, err := sql.Open("postgres", connStr)
+func executeQuery(db *sql.DB, query string, args ...interface{}) (int64, error) {
+	result, err := db.Exec(query, args...)
 	if err != nil {
 		return 0, err
 	}
-	defer db.Close()
 
-	result, err := db.Exec(query, args...)
+	return result.RowsAffected()
+}
+
+// Initialise opens the database connection
+func (dao *DAO) Initialise(configFilePath string) error {
```

Comment:
I’d be tempted to pass in the config here and do the file IO in main. Keeps all the side effects in one place and should be easier to test if we want to